src/Logger.php

```diff
@@ -239,13 +236,13 @@ protected function dispatch($messages, bool $final): void
     {
         $targetErrors = [];
         foreach ($this->getTargets() as $target) {
-            if ($target->enabled) {
+            if ($target->getEnabled()) {
                 try {
                     $target->collect($messages, $final);
                 } catch (\Exception $e) {
-                    $target->enabled = false;
+                    $target->setEnabled(false);
```

Comment:
Method could be split into `enable()` and `disable()`.

Reply:
Added `enable()` & `disable()` but kept `setEnabled()` for the callback form